internal/pkg/cli/secret_init.go

```diff
@@ -75,20 +98,97 @@ func (o *secretInitOpts) Validate() error {
 }
 
 func (o *secretInitOpts) Ask() error {
+	if o.overwrite {
+		log.Infof("You have specified %s flag. Please note that overwriting an existing secret may break your deployed service.\n", color.HighlightCode("--overwrite"))
+	}
+	if err := o.askForAppName(); err != nil {
+		return err
+	}
+	if err := o.askForSecretName(); err != nil {
+		return err
+	}
+	if err := o.askForSecretValues(); err != nil {
+		return err
+	}
 	return nil
 }
 
 func (o *secretInitOpts) Execute() error {
 	return nil
 }
 
+func (o *secretInitOpts) askForAppName() error {
+	if o.appName != "" {
+		return nil
+	}
+
+	app, err := o.selector.Application(secretInitAppPrompt, secretInitAppPromptHelp)
+	if err != nil {
+		return fmt.Errorf("ask for an application to add the secret to: %w", err)
+	}
+	o.appName = app
+	return nil
+}
+
+func (o *secretInitOpts) askForSecretName() error {
+	if o.name != "" {
+		return nil
+	}
+
+	name, err := o.prompter.Get(secretInitSecretNamePrompt,
+		secretInitSecretNamePromptHelp,
+		validateSecretName,
+		prompt.WithFinalMessage("secret name: "))
+	if err != nil {
+		return fmt.Errorf("ask for the secret name: %w", err)
+	}
+
+	o.name = name
+	return nil
+}
+
+func (o *secretInitOpts) askForSecretValues() error {
+	if o.values != nil {
+		return nil
+	}
+
+	envs, err := o.store.ListEnvironments(o.appName)
+	if err != nil {
+		return fmt.Errorf("list environments in app %s: %w", o.appName, err)
+	}
+
+	if len(envs) == 0 {
+		log.Errorf("Secrets are environment-level resource. Please run %s before running %s.\n",
+			color.HighlightCode("copilot env init"),
+			color.HighlightCode("copilot secret init"))
+		return fmt.Errorf("no environment is found in app %s", o.appName)
+	}
+
+	values := make(map[string]string)
+	for _, env := range envs {
+		value, err := o.prompter.GetSecret(
```

Comment:
What do you think of making default from the second env on to have the default value as the previous one? The first one's default value would be just ""

Reply:
I like the idea! This way folks won't have to enter the same value repeatedly. I have one concern though.

Currently, if the user press 'enter' without giving any values, we simply won't create that secret for them in that environment. This information is included in the prompt help: `fmtSecretInitSecretValuePromptHelp = "If you do not wish to add the secret %s to environment %s, you can leave this blank by pressing 'enter' without entering any value."`

If we auto-fill with default value, I think there will be no way for people to say "I don't want this secret in this environment" with the current prompting scheme. This will introduce a gap between guided experience and the flags - with `--values` you can obviously not create the secret in some environments, however with promptings it'd become impossible.

Reply:
Ah I see. That makes sense to me.